src/pattern-library/components/patterns/data/DataTablePage.tsx

```diff
@@ -1,17 +1,87 @@
-import { useState } from 'preact/hooks';
+import { useCallback, useMemo, useState } from 'preact/hooks';
 
-import { DataTable, Scroll } from '../../../../';
+import { DataTable, type DataTableProps, Scroll } from '../../../../';
+import type { Order } from '../../../../components/data/DataTable';
 import Library from '../../Library';
 import { nabokovNovels } from '../samples';
 import type { NabokovNovel } from '../samples';
 
 const nabokovRows = nabokovNovels();
 const nabokovColumns = [
-  { field: 'title', label: 'Title' },
-  { field: 'year', label: 'Year' },
-  { field: 'language', label: 'Language' },
+  { field: 'title' as const, label: 'Title' },
+  { field: 'year' as const, label: 'Year' },
+  { field: 'language' as const, label: 'Language' },
 ];
 
+type StrictNabokovNovel = Omit<NabokovNovel, 'translatedTitle'>;
+
+function useOrderedRows(
+  rows: StrictNabokovNovel[],
+  order: Order<keyof StrictNabokovNovel>,
+) {
+  return useMemo(() => {
+    if (!order) {
+      return rows;
+    }
+
+    return [...rows].sort((a, b) => {
+      if (a[order.field] === b[order.field]) {
+        return 0;
+      }
+
+      if (order.direction === 'ascending') {
+        return a[order.field] > b[order.field] ? 1 : -1;
+      }
+
+      return a[order.field] > b[order.field] ? -1 : 1;
+    });
+  }, [order, rows]);
+}
```

Comment:
I reckon we might want to eventually extract this hook and expose it as a utility.